mlx/ops.h

```diff
@@ -1105,4 +1105,13 @@ array addmm(
     const float& beta = 1.f,
     StreamOrDevice s = {});
 
+/**
+ * Implements the identity function but allows injecting dependencies to other
+ * arrays. This ensures that these other arrays will also have been computed
+ * when the outputs of this function are computed.
+ */
+std::vector<array> depends(
```

Comment:
I might be wrong about this, but it looks like we could use `depends` as a replacement for `Synchronizer`?

Reply:
Yes we can. Since we are creating the array manually, we don't even need to have an output. Nobody ever calls eval on the Synchronizer anyway.